langserver/handlers/initialize.go

```diff
@@ -118,7 +131,7 @@ func (lh *logHandler) Initialize(ctx context.Context, params lsp.InitializeParam
 	// Walker runs asynchronously so we're intentionally *not*
 	// passing the request context here
 	bCtx := context.Background()
-	err = walker.StartWalking(bCtx, fh.Dir(), func(ctx context.Context, dir string) error {
+	err = walker.StartWalking(bCtx, fh.Dir(), excludeModulePaths, func(ctx context.Context, dir string) error {
```

Comment:
making excludeModulePaths a property of `walker` instance will make it have state. since `startWalking` func will only be invoked once, there won't be problems. I will do it in this way, but we might take care of it because it might lead unexpected results if invoking multiple times.

Reply:
FWIW the walker isn't quite ready to be invoked more than once now either way (due to some other internal state), but I think that making it a property will make it easier later for walking and configuration resetting to be managed independently. At that point we can guard the slice with a mutex, or maybe a semaphore, so that e.g. we can detect when user attempts to change the configuration while the walker is walking and decide (or let them decide) what to do.